File: examples/golang/data/initial/android.go
// +build test

package initial

import (
	"forcedata/examples/golang/model"
)

// データの識別子
const (
	Test01Android1 model.DataID = "Test01Android1"
)

func RegisterAndroid() {
	f := model.Factory
	f.Register(Test01Android1, func() interface{} {
		return &model.Android{
			Manufacturer: "google",
			Model:        "pixel5",
		}
	})
}

File: examples/typescript/data/initial/android.ts
import { factory } from '../../model/factory';
import { Android } from '../../model/model';
import { DATAID } from '../dataid';

export function registerInitialAndroid() {
  const f = factory;
  f.register(DATAID.TEST01_ANDROID_1, () => {
    return {
      manufacturer: 'google',
      model: 'pixel5',
    } as Android;
  });
}
